migration/orm.go

```diff
@@ -0,0 +1,66 @@
+package migration
+
+import (
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/errors"
+	"github.com/iov-one/weave/orm"
+)
+
+type Bucket struct {
+	orm.Bucket
+	packageName string
+	schema      *SchemaBucket
+	migrations  *register
+}
+
+func NewBucket(packageName string, bucketName string, model orm.Cloneable) Bucket {
+	return Bucket{
+		Bucket:      orm.NewBucket(bucketName, model),
+		packageName: packageName,
+		schema:      NewSchemaBucket(),
+		migrations:  reg,
```

Comment:
Ah, so you copy a reference here to the global register with the purpose of overriding it for test cases, right? There is no other need to copy this reference.

But it does seem like a good way to handle mocking. Maybe add a function to set it, so tests read clearer.

`func (svb *Bucket) mockMigrations()` -> `svb.migrations = newRegistry()`

And then just register migrations on the bucket with a public helper (to global namespace unless mocked for tests):

`func (svb *Bucket) MustRegister(...)` -> `svb.migrations.MustRegister(...)`

One can also use this in the init functions. Maybe even pulling the model automatically from the bucket (in a future PR)

Reply:
Done in https://github.com/iov-one/weave/pull/550/commits/88270383a81e3290ce2993b3fea9beaa8579bfc5